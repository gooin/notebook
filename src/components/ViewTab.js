import React from 'react';
import {Tabs} from 'antd';
import {CHART_VIEW, LIST_VIEW, CHART_MODE, LIST_MODE} from '../utility'
import {UnorderedListOutlined, PieChartOutlined} from '@ant-design/icons';
import PropTypes from 'prop-types';

const {TabPane} = Tabs;


const ViewTab = ({activeTab, onTabViewChange}) => (
    <Tabs type="card"
        // activeKey={activeTab}
        //   onChange={(key) => {
        //       console.log("tabs changed.", key);
        //   }}
          onChange={onTabViewChange}
    >
        <TabPane tab={
            <span>
            <UnorderedListOutlined/>
                {LIST_MODE}
            </span>
        }
                 key={LIST_VIEW}
        >
            {/*Content of Tab Pane {LIST_VIEW}*/}
        </TabPane>
        <TabPane tab={
            <span>
            <PieChartOutlined/>
                {CHART_MODE}
        </span>
        }
                 key={CHART_VIEW}
        >
            {/*Content of Tab Pane {CHART_VIEW}*/}
        </TabPane>
    </Tabs>
);

ViewTab.propType = {
    onTabChange: PropTypes.func.isRequired
}


export default ViewTab;