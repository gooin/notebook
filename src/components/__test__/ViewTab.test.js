import React from 'react';
import {shallow} from 'enzyme';
import ViewTab from "../ViewTab";
import toJson from 'enzyme-to-json';


const props = {
    onTabViewChange: jest.fn()
}


describe('test ViewTab component',()=>{
    let wrapper = shallow(<ViewTab {...props}/>)
    it('should Tabs rendered',()=>{
        expect(wrapper.find('TabPane').length).toBe(2)
    })

    it('check snapshot',()=>{
        expect(toJson(wrapper)).toMatchSnapshot()
    })

})