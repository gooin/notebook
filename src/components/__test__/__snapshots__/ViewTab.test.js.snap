// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test ViewTab component check snapshot 1`] = `
<Tabs
  onChange={[MockFunction]}
  type="card"
>
  <TabPane
    key="list"
    tab={
      <span>
        <ForwardRef(UnorderedListOutlined) />
        列表模式
      </span>
    }
  />
  <TabPane
    key="chart"
    tab={
      <span>
        <ForwardRef(PieChartOutlined) />
        图表模式
      </span>
    }
  />
</Tabs>
`;
